### DISCUSSION TITLE
false running process prompt on vertical splits

### DISCUSSION THREAD

**Discussion by umutbasal** [ghostty-org/ghostty#3057] [Q&A] [answered]
https://github.com/user-attachments/assets/2e4449d0-ad79-4a59-815f-a2acaeb1121b

version
```
Ghostty 0.1.0-main+05fd5e0c

Version
  - version: 0.1.0-main+05fd5e0c
  - channel: tip
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
```
fresh install no config

**Comment by umutbasal** [comment]
Ghostty 1.0.1-main+35b9ceee still have it.
I've added `confirm-close-surface = false` to my cfg temporarily. Somehow close_surface doesnt get process alive correctly with zsh. didn't try with other shells.

**Reply by goonzoid** [reply]
Are you using https://github.com/sindresorhus/pure for your prompt? If so, it runs an additional background zsh process that used to cause a similar issue on iTerm2. Worth updating to the latest version of pure if you're not already on it, since I believe they made a change to fix it.

**Reply by rhodes-b** [reply]
Your shell in general could also be considered a process saying not all the virtical splits prompted for closing, most notibly the very first one you closed, in fact I see this on both virtual and horizontal splits for the popup

**Comment by 00-kat** [comment]
Possibly related to #3256.

**Comment by mitchellh** [comment, marked as answer]
We haven't heard further reports since Ghostty 1.2. This issue is very old. If this issue is still happening in specific cases, please open a new discussion.
